feat: detect and support window.ipfs.enable experiment

lidel opened the pull request:
Just to get the ball rolling, this PR adds a simple heuristic for forward-compatible detection of IPFS API exposed by experimental branch of ipfs-companion: https://github.com/ipfs-shipyard/ipfs-companion/issues/589

@alanshaw I can take care of releasing this if you make me a colaborator on NPM.

lidel committed:
feat: detect and support window.ipfs.enable

This is a simple heuristic for forward-compatible detection of IPFS API
exposed by ipfs-companion

License: MIT
Signed-off-by: Marcin Rataj <lidel@lidel.org>

alanshaw commented on index.js:
```diff
@@ -2,7 +2,15 @@ module.exports = function getIpfs (opts) {
+      // forward-compatible migration
+      // https://github.com/ipfs-shipyard/ipfs-companion/issues/589
+      if (typeof window.ipfs.enable === 'function') {
+        return resolve(window.ipfs.enable())
```
Pass wanted permissions...so we don't have to do another release??

```suggestion
        return resolve(window.ipfs.enable(opts.permissions))
```

alanshaw reviewed (changes requested):
Code LGTM - would you mind adding a test?

lidel committed:
feat: support customizing window.ipfs defaults

While we are at it, make it possible to request specific permissions
and in future other capabilities provided by this interface.

License: MIT
Signed-off-by: Marcin Rataj <lidel@lidel.org>

lidel commented:
@alanshaw done, check now:
- added test
- added support for overriding `defaults`
  - did not want to name it 'permissions` as we may want to pass other flags

alanshaw commented:
> - added support for overriding `defaults`

hmm, I don't expect to pass "defaults" normally! It suggests something is going to override them...when actually what I pass would usually override anything set by default.

>   - did not want to name it 'permissions` as we may want to pass other flags 

Unless you have something in mind right now I'd suggest sticking to `permissions` (we know we want that) and change it when we want to pass something else/more.

lidel committed:
style: defaults → permissions

License: MIT
Signed-off-by: Marcin Rataj <lidel@lidel.org>

lidel commented:
@alanshaw lets go with `permissions` for now (https://github.com/ipfs-shipyard/window.ipfs-fallback/pull/2/commits/bed7c7206f0dd8ffaee18ada8f5bfd4bc49cab03)
